platforms/ide/ide-plugins/src/main/java/org/gradle/plugins/ide/internal/tooling/IdeaModelBuilder.java

```diff
@@ -70,6 +70,7 @@ public DefaultIdeaProject buildAll(String modelName, Project project) {
     @Override
     public DefaultIdeaProject buildForRoot(Project project, boolean offlineDependencyResolution) {
         Project root = project.getRootProject();
+        ToolingModelDeprecations.nagOnNonRootTarget(project, root);
```

Comment:
We can probably do it for all model builders, not just the build-in ones you selected. 
I assume this is to do some initial testing and the general approach will come later.

Reply:
This shouldn't be applicable to all model builders, only to the select ones we know and maintain.

There are many model builders that make sense in the scope of a given project. For instance, `BuildInvocations` is project-specific, because it provide info about the tasks in the given project (and its subprojects...). I am certain that there are 3rd party builders like the ones from IDEA that are also project-scoped.

This PR is concerned with specific model builders which pretend to be project-specific, but instead silently and unconditionally switch the target to be the root project.

Does this clarify the intention, or do you think we shouldn't proceed with this deprecation then?

Reply:
I had the impression that you plan to enforce the same thing globally. Thanks for the clarification, I'm good with the changes then.